uaclient/daemon/poll_for_pro_license.py

```diff
@@ -46,8 +47,10 @@ def poll_for_pro_license(cfg: UAConfig):
         LOG.debug("Not on cloud, shutting down")
         return
 
-    if not isinstance(cloud, UAAutoAttachGCPInstance):
-        LOG.debug("Not on gcp, shutting down")
+    if not isinstance(cloud, UAAutoAttachGCPInstance) and not isinstance(
+        cloud, UAAutoAttachAzureInstance
+    ):
+        LOG.debug("Not on supported cloud platform, shutting down")
```

Comment:
I think for this feature we would want any unexpected exceptions to result in a crash so that we would get a crash report. I think we're covering the main expected exception of HTTPError, and anything else would be a big-enough problem to warrant a crash

Reply:
+1 no need to except Exception